cachi2/core/package_managers/yarn/main.py

```diff
@@ -86,9 +88,11 @@ def _fetch_dependencies(source_dir: RootedPath, output_dir: RootedPath) -> None:
     :param output_dir: the directory where the yarn dependencies will be downloaded to.
     :raises CalledProcessError: if the 'yarn install' command fails.
     """
-    # the output_dir is where the globalFolder must be set to
-    # the yarn commands can be called by using the core.utils.run_cmd function
-    pass
+    args = ["install", "--mode", "skip-build"]
+    try:
+        run_yarn_cmd(args, source_dir, {"YARN_GLOBAL_FOLDER": str(output_dir)})
+    except subprocess.CalledProcessError:
+        raise
```

Comment:
Gomod has a specific [error](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/errors.py#L99) type for subprocess errors? Do we want a similar one for Yarn?

Reply:
IMO that would mainly depend on whether cachi2 wants to wrap different Yarn error codes which we'd have to pull from subprocess, IOW looking at the Gomod error, it's just a generic Cachi2Error which prints out some string and optionally provides a string formatted solution, so technically a generic Cachi2Error would be enough, but since there's already an existing precedent, you have a strong point.

Reply:
Yeah, at least we want to re-raise this as some subclass of Cachi2Error, a new one for Yarn probably makes sense